cmd/orchestrator/main.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+
+	activatorhandler "github.com/knative/serving/pkg/activator/handler"
+	ohandler "github.com/kubeflow/kfserving/pkg/orchestrator/handler"
+	perrors "github.com/pkg/errors"
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/h2c"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	serviceFQDN = flag.String("service", "", "The FQDN of the service to proxy")
+)
+
+func main() {
+	flag.Parse()
+
+	logf.SetLogger(logf.ZapLogger(false))
+	log := logf.Log.WithName("entrypoint")
+
+	stopCh := signals.SetupSignalHandler()
+
+	// Create handler chain
+	// Note: innermost handlers are specified first, ie. the last handler in the chain will be executed first
+	var oh http.Handler = ohandler.New(log, *serviceFQDN)
+	oh = &activatorhandler.ProbeHandler{NextHandler: oh}
```

Comment:
I don' think we want to take a dependency on this. Can you elaborate on why we're tightly coupling ourselves to their proxy?

Reply:
I agree. The handlers were examples but actually not sure we need them so there shouldn't be a tight dependency. We have these packages in our deps so simply wanted to reuse any code they have already done.

Reply:
Ok, makes sense. I'm not super clear on the chain of handlers logic. Does every handler handle every request? Should there be some path based logic?